tendrl-dashboard.spec

```diff
@@ -8,16 +8,57 @@ Group:   Applications/System
 Source0: %{name}-%{version}.tar.gz
 URL: https://github.com/Tendrl/dashboard
 
-Requires: tendrl-api
+BuildRequires: npm
+BuildRequires: nodejs-packaging
+BuildRequires: fontconfig
 
 %description
 Contains the JavaScript GUI content for the Tendrl front-end components
 (dashboard, login screens, administration screens)
 
 %prep
-%autosetup
+%setup -q -n %{name}-%{version}
+%nodejs_fixdep foomodule
 
 %build
+npm install gulp
```

Comment:
Yes, I think there is going to have to be at least some pre-compilation that happens in the release tarball before it gets to dist-git/Koji.

It would be a good idea to ask other packagers who maintain large Javascript packages. For example, I looked through Cockpit's source and it appears that they compile all of the Javascript into the single cockpit-134.tar.xz release tarball before they push it to GitHub. Compare their [bower_components/*](https://github.com/cockpit-project/cockpit/tree/master/bower_components) files in Git and the [tarball](https://github.com/cockpit-project/cockpit/releases/download/134/cockpit-134.tar.xz).

cockpit-devel@lists.fedorahosted.org may have more suggestions.